docs/current_docs/manuals/developer/python.mdx

```diff
@@ -0,0 +1,98 @@
+---
+slug: /manuals/developer/python
+displayed_sidebar: "current"
+toc_max_heading_level: 2
+title: "Python Extras"
+---
+
+# Python Extras
+
+This page lists features specific to the Python SDK.
+
+## Name overrides
+
+Sometimes it's possible to get into a situation where the name you want to use
+for the Dagger API is reserved or has a naming conflict within Python.
+
+To resolve this, the [`@dagger.function`][@function] decorator and the
+[`dagger.field`][dag-field] descriptor have a `name` argument to provide
+an alternative name for the Dagger API. And for function arguments,
+[annotate][Annotated] with an additional [`dagger.Arg`][dag-arg] metadata.
+
+Here's an example with a renamed object attribute (and constructor argument),
+function and function argument:
+
+```python file=./snippets/python/name-overrides/main.py
+```
+
+Confirm with `dagger call --help` that the names were overridden:
+
+```
+FUNCTIONS
+  def           Definition
+  import        Import the specified image
+
+ARGUMENTS
+      --def string   Definition (default "latest")
+```
+
+And the same for the function argument, with `dagger call import --help`:
+
+```
+ARGUMENTS
+      --from string   Image ref (default "alpine")
+```
+
+[@function]: https://dagger-io.readthedocs.io/en/latest/module.html#dagger.function
+[dag-field]: https://dagger-io.readthedocs.io/en/latest/module.html#dagger.field
+[dag-arg]: https://dagger-io.readthedocs.io/en/latest/module.html#dagger.Arg
+[Annotated]: https://docs.python.org/3/library/typing.html#typing.Annotated
+
+## Python debug logs
+
+The Python SDK prints debugging information at various steps of the execution
+flow of a module, which can be very useful in understanding what's being
+received from and sent to the API.
+
+This is done using standard Python [logging](https://docs.python.org/3/howto/logging.html),
+so it's highly configurable (for example, saving to a file or sending to an
+external system like Sentry). But for a simple lowering of the default logging
+level to [logging.DEBUG](https://docs.python.org/3/library/logging.html#logging.DEBUG),
+there's a convenience function for that:
+
+```python file=./snippets/python/debugging/main.py
+```
+
+:::note
+With the TUI, you need to use a progress output that doesn't collapse on success
+like `--progress=plain` or `--debug`, otherwise it won't show in the terminal.
+:::
+
+Using the command `dagger call --debug echo --msg="hello"`, should print
+something like:
+
+```
+✔ MyModule.echo(msg: "hello"): String! 0.5s
```

Comment:
@helderco the output now contains a very large number of spans, and I wasn't able to find anything similar to what is shown here. Please suggest what should be done for this output.